go/vt/vttablet/onlineddl/executor.go

```diff
@@ -2535,6 +2536,23 @@ func (e *Executor) updateMigrationProgress(ctx context.Context, uuid string, pro
 	return err
 }
 
+func (e *Executor) updateRowsCopied(ctx context.Context, uuid string, rowsCopied int64) error {
+	if rowsCopied <= 0 {
+		// Number of rows can only be positive. Zero or negative must mean "no information" and
+		// we don't update the table value.
+		return nil
+	}
+	query, err := sqlparser.ParseAndBind(sqlUpdateMigrationRowsCopied,
```

Comment:
My only concern is that this `ParseAndBind` could be optimized by only parsing the statement once and re-binding on each update, but I see the same pattern is used throughput the executor, so optimizing only for `updateRowsCopied` seems wrong. I'll revisit this down the road and refactor the whole file.

Reply:
This statement is expected to run once or twice a minute. All over the Online DDL code, I don't try to optimize query performance. Do feel free to rewrite if you want to have uniform coding style through the code, but optimization-wise this is not an interesting IMHO.